types/src/contract_event.rs

```diff
@@ -232,25 +289,31 @@ impl std::fmt::Debug for ContractEventV1 {
 }
 
 /// Entry produced via a call to the `emit` builtin.
-#[derive(Hash, Clone, Eq, PartialEq, Serialize, Deserialize, CryptoHasher)]
+#[derive(Hash, Clone, Eq, PartialEq, Serialize, CryptoHasher)]
 pub struct ContractEventV2 {
     /// The type of the data
     type_tag: TypeTag,
     /// The data payload of the event
     #[serde(with = "serde_bytes")]
     event_data: Vec<u8>,
+    /// Cached event size used for gas charging. This field is not included when serialized and
+    /// on deserialization is recomputed (via custom deserializer).
+    #[serde(skip_serializing, skip_deserializing)]
+    size: usize,
```

Comment:
safer to just remove deriving Serialize and manually implement

Reply:
done: manually implemented